conftest.py

```diff
@@ -17,6 +17,78 @@ def pytest_configure(config):
     warnings.filterwarnings("error", "", Warning, r"^(?!(|kombu|raven|sentry))")
 
 
+def enable_file_tracking():
+    import builtins
+    import socket
+    import traceback
+    import weakref
+
+    from _pytest import unraisableexception
+
+    original_open = open
+    original_socket = socket.socket
+    handle_stacks = {}
+
+    def _handle_key(handle):
+        # reach through the handle to the innermost piece
+        if hasattr(handle, "buffer") and hasattr(handle.buffer, "fileno"):
+            handle = handle.buffer
+        if hasattr(handle, "raw") and hasattr(handle.raw, "fileno"):
+            handle = handle.raw
+        return (id(handle), handle.fileno())
+
+    class patched_catch_unraisable_exception(unraisableexception.catch_unraisable_exception):
```

Comment:
I believe a saner way to monkeypatch is to just directly patch `_hook` on the baseclass. at least that's what we usually do in the sdk

otherwise a check for `isinstance(foo, patched_catch_unraisable_exception)` might fail because the `foo` instance is actually of the old, unpatched kind. also already-defined subtypes of `unraisableexception` will not get the new hook impl

but not entirely sure. i also have no clue what this class does